### PR TITLE
[SPARK-17438][WebUI]	Show Application.executorLimit in the application page

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

This PR adds `Application.executorLimit` to the applicatino page
## How was this patch tested?

Checked the UI manually.

Screenshots:
1. Dynamic allocation is disabled

screen shot 2016-09-07 at 4 21 49 pm
1. Dynamic allocation is enabled.

screen shot 2016-09-07 at 4 25 30 pm

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala [hunk 1]
```diff
@@ -61,11 +68,11 @@ private[ui] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app")
             <li><strong>User:</strong> {app.desc.user}</li>
             <li><strong>Cores:</strong>
             {
-              if (app.desc.maxCores.isEmpty) {
+              if (maxCores.isEmpty) {
                 "Unlimited (%s granted)".format(app.coresGranted)
               } else {
                 "%s (%s granted, %s left)".format(
-                  app.desc.maxCores.get, app.coresGranted, app.coresLeft)
+                  maxCores.get, app.coresGranted, maxCores.get - app.coresGranted)
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala]: note: `maxCores.get - app.coresGranted` may be negative before we kill the executors.

**Comment by zsxwing** [comment]
cc @tdas

**Comment by zsxwing** [comment]
@JoshRosen do you have time to review this PR?

**Comment by JoshRosen** [comment]
Minor typo in the PR title and description: applicatino -> application

File: core/src/main/scala/org/apache/spark/ui/ToolTips.scala [hunk 2]
```diff
@@ -90,4 +90,10 @@ private[spark] object ToolTips {
 
   val TASK_TIME =
   "Shaded red when garbage collection (GC) time is over 10% of task time"
+
+  val APPLICATION_EXECUTOR_LIMIT =
+    """The number of executors this application can have at any given time. This is set only when
```
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/ui/ToolTips.scala]: How about something like

> Maximum number of executors that this application will use. This limit is finite only when dynamic allocation is enabled.

File: core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala [hunk 3]
```diff
@@ -70,6 +70,16 @@ private[ui] class ApplicationPage(parent: MasterWebUI) extends WebUIPage("app")
             }
             </li>
             <li>
+              <span data-toggle="tooltip" title={ToolTips.APPLICATION_EXECUTOR_LIMIT}
+                    data-placement="right">
+                <strong>Executor Limit: </strong>
+                {
+                  if (app.executorLimit == Int.MaxValue) "Unlimited" else app.executorLimit
```
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala]: I noticed that `ApplicationDescription` has a `maxCores` field which is displayed on the Master UI's applications page. Do you think that there's any possibility for confusion between the limits displayed there (which are measured in terms of cores) and the executor limits shown here?

**Review by JoshRosen — COMMENTED** [review]
This change looks good to me overall but I have one suggestion regarding wording and a question regarding the `maxCores` limit shown on the Master UI.

File: core/src/main/scala/org/apache/spark/ui/ToolTips.scala (hunk 2 above)
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/ui/ToolTips.scala]: Updated

File: core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala (hunk 3 above)
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala]: Agreed. Actually, there is another confusing difference between executor limit and maxCores: `maxCores` is fixed for an application while `executorLimit` is mutable. I'm not sure how to clarify them just on UI.

Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/ApplicationPage.scala]: I think for now it's OK. Conceptually the number of executors is constrained by the lower of the two: the executor limit and the cores limit (when translated into executors).

**Comment by andrewor14** [comment]
LGTM, merging into master 2.0 thanks @zsxwing!

**Pull request closed without merging**